include/seqan3/io/alignment_file/format_bam.hpp

```diff
@@ -56,63 +56,26 @@ namespace seqan3
  *
  * \copydetails seqan3::format_sam
  */
-struct format_bam
+class format_bam : private detail::format_sam_base
 {
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    format_bam() = default; //!\brief Defaulted.
```

Comment:
Also add `noexcpt`

Reply:
I thought explicitily defaulted constructor are by default noexcept?

Reply:
btw. `format_sam` cannot be noexcept default constructible  because it has a string buffer which is not `noexcept` default constructible..

Reply:
> I thought explicitily defaulted constructor are by default noexcept?

Yes, but writing noexcept generates a diagnostic if it cannot be made noexcept. It's a good way to check.

> btw. format_sam cannot be noexcept default constructible because it has a string buffer which is not noexcept default constructible..

Does the string buffer need to be dynamic? Then that's OK! 
If you know it will always be less than e.g. 1000, I would use a seqan3::small_string instead. Although that would make objects expensive to move :thinking: 
Anyway, up to you. Maybe add a source comment why it's not noexcept.

Reply:
> Yes, but writing noexcept generates a diagnostic if it cannot be made noexcept. It's a good way to check.

True, but I think you are the only one watching out for this. I see a lot of places were noexcept was not explicitly used... I always feel like we should be consistent in our reviews, thats why I tend to put these types of "unimportant" discussion points to our list for the strategy meeting.. but it is probably not worth the discussion

> Does the string buffer need to be dynamic? Then that's OK!

yes it is for the quality string. Usually quality strings in SAM are small but you never know :woman_shrugging: 

I'll add a source comment